Quake/gl_screen.c

```diff
@@ -698,6 +700,47 @@ void SCR_DrawClock (cb_context_t *cbx)
 	}
 }
 
+/*
+==============
+SCR_DrawSpeed
+==============
+*/
+void SCR_DrawSpeed (cb_context_t *cbx)
+{
+	if (cl.intermission || !scr_showspeed.value)
+		return;
+
+	float maxSpeedFillWidth = 600; // if speed >= val, then speedometer is completely filled
+	float scale = scr_sbarscale.value;
+	float width = 140.f * scale;
+	float height = 12.f * scale;
+	float x = ((float)vid.width / 2.f) - (width / 2.f);
+	float y = (float)vid.height - height - ((float)Sbar_HudHeight () * scale);
+
+	float textLeftOffset = 10.f * scale;
+	float textHeight = 8.f * scale;
+	float textY = y + (height / 2) - (textHeight / 2);
+	float textX = x + textLeftOffset;
+
+	char  st[4];
+	float speed = VectorLength (cl.velocity);
+	sprintf (st, "%-3d", (int)speed);
+	int speedWidth = (int)(fmin (1.f, speed / maxSpeedFillWidth) * width);
+
+	static int bgColor = -1;
+	static int fillColor = -1;
+	if (bgColor == -1)
+	{
+		bgColor = TexMgr_NearestColor (20, 20, 0);
```

Comment:
What's the point of this? Can't you just hard code the closest color from the palette? The palette is fixed as far as I know? Why does it even have to be a palette color?